### PR TITLE
Fix releases, includes_releases, and a few others to use Liquid link syntax

### PR DESCRIPTION
**Pull request by mdlinville**
Fix releases, includes_releases, and a few others to use Liquid link syntax

Port of https://github.com/cockroachdb/docs/pull/18848 to cloud-2.0 branch

Partially resolves DOC-10953

### CODE REVIEW

File: src/current/releases/cloud.md [hunk 1]
```diff
@@ -188,7 +186,7 @@ Get future release notes emailed to you:
 
 <h3 id="2024-01-17-general-updates"> General updates </h3>
 
-- The single-page CockroachDB Cloud [Create cluster]({% link cockroachcloud/create-a-serverless-cluster.md %}) and [Edit cluster]({% link cockroachcloud/basic-cluster-management.md %}) have been updated to use multi-step wizards.
+- The single-page CockroachDB Cloud [Create cluster]({% link cockroachcloud/create-a-basic-cluster.md %}) and [Edit cluster]({% link cockroachcloud/basic-cluster-management.md %}) have been updated to use multi-step wizards.
```
Comment (taroface) [inline src/current/releases/cloud.md:189]: Should "Cloud" be using the liquid var?

File: src/current/v23.1/start-a-local-cluster.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ The store directory is `cockroach-data/` in the same directory as the `cockroach
 
 ## Step 1. Start the cluster
 
-This section shows how to start a cluster interactively. In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link v23.1/deploy-cockroachdb-on-premises.md %}?filters=systemd).
+This section shows how to start a cluster interactively. In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link {{ PAGE.VERSION.VERSION }}/deploy-cockroachdb-on-premises.md %}?filters=systemd).
```
Comment (taroface) [inline src/current/v23.1/start-a-local-cluster.md:29]:
```suggestion
This section shows how to start a cluster interactively. In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link {{ page.version.version }}/deploy-cockroachdb-on-premises.md %}?filters=systemd).
```

File: src/current/v23.1/start-a-local-cluster.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ This section shows how to start a cluster interactively. In production, operator
     {{site.data.alerts.callout_info}}
     The `--background` flag is not recommended. If you decide to start nodes in the background, you must also pass the `--pid-file` argument. To stop a `cockroach` process running in the background, extract the process ID from the PID file and pass it to the command to [stop the node](#step-7-stop-the-cluster).
 
-    In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link v23.1/deploy-cockroachdb-on-premises.md %}?filters=systemd).
+    In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link {{ PAGE.VERSION.VERSION }}/deploy-cockroachdb-on-premises.md %}?filters=systemd).
```
Comment (taroface) [inline src/current/v23.1/start-a-local-cluster.md:46]:
```suggestion
    In production, operators usually use a process manager like `systemd` to start and manage the `cockroach` process on each node. Refer to [Deploy CockroachDB On-Premises]({% link {{ page.version.version }}/deploy-cockroachdb-on-premises.md %}?filters=systemd).
```

**Review by taroface — APPROVED** [review]
LGTM assuming the broken links are fixed.

File: src/current/releases/cloud.md (hunk 1 above)
Comment (mdlinville) [reply src/current/releases/cloud.md:189]: We don't require it really, it doesn't provide much benefit. We actually went away from it for a while and then seem to have begun using it again. I'm neutral on it and will leave it alone here.

File: src/current/v23.1/start-a-local-cluster.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/v23.1/start-a-local-cluster.md:29]: Wow I wonder how that happened!

**Pull request merged**